src/ucs/sys/sys.c

```diff
@@ -330,25 +330,41 @@ size_t ucs_get_page_size()
     return page_size;
 }
 
-size_t ucs_get_huge_page_size()
+size_t ucs_get_meminfo_entry(char *pattern)
```

Comment:
would be better to pass the **name** (e.g "MemFree", "Hugepagesize") instead of pattern - to avoid code duplication. since all entries in meminfo have same pattern anyway.
also, could multiply by 1024 inside this function too.